xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -260,18 +261,29 @@ void CLinuxRendererGLES::LoadPlane(CYuvPlane& plane, int type,
   glBindTexture(m_textureTarget, plane.id);
 
   // OpenGL ES does not support strided texture input.
+  GLint pixelStore = -1;
   if (stride != static_cast<int>(width * bps))
   {
-    unsigned char* src = static_cast<unsigned char*>(data);
-    for (unsigned int y = 0; y < height; ++y, src += stride)
+    unsigned int verMajor, verMinor;
+    m_renderSystem->GetRenderVersion(verMajor, verMinor);
+
+    if (verMajor < 3)
     {
-      glTexSubImage2D(m_textureTarget, 0, 0, y, width, 1, type, GL_UNSIGNED_BYTE, src);
+      unsigned char* src (static_cast<unsigned char*>(data)), *dst(m_planeBuffer);
+      for (unsigned int y = 0; y < height; ++y, src += stride, dst += width)
+        memcpy(dst, src, width);
+      pixelData = m_planeBuffer;
+    }
+    else
+    {
+      glGetIntegerv(GL_UNPACK_ROW_LENGTH, &pixelStore);
+      glPixelStorei(GL_UNPACK_ROW_LENGTH, stride);
```

Comment:
`GL_UNPACK_ROW_LENGTH` requires GLES3 headers, please protect usage with `#if HAS_GLES == 3` or `#ifdef GL_UNPACK_ROW_LENGTH`, see https://github.com/Kwiboo/xbmc/commit/b6f23d89b2fdaa83d280d4f7b13aadc506c52f37 for a possible fixup, fixes building when there is no GLES3 include headers.

Reply:
Line 270

Reply:
that only check runtime for gles 3 support, the issue is at compile time where `GL_UNPACK_ROW_LENGTH` is undefined on e.g. Rockchip RK3328 platform as no GLES3 include files is shipped, also `GLES3/gl3.h` is only included if `HAS_GLES == 3` at https://github.com/xbmc/xbmc/blob/master/xbmc/system_gl.h#L36-L38

Reply:
Ok, understand, thx